functions/Get-DbaDiskSpace.ps1

```diff
@@ -137,57 +175,57 @@ Returns a custom object filled with information for server1, server2 and server3
 						}
 					}
 				}
-				
-				if (!$diskname.StartsWith("\\"))
+
+				if (!$diskname.StartsWith("\\") -or $Detailed)
 				{
-					$total = "{0:n2}" -f ($disk.Capacity/$measure)
-					$free = "{0:n2}" -f ($disk.Freespace/$measure)
-					$percentfree = "{0:n2}" -f (($disk.Freespace / $disk.Capacity) * 100)
-					
-					if ($CheckForSql -eq $true)
-					{
-						$alldisks += [PSCustomObject]@{
-							Server = $server
-							Name = $diskname
-							Label = $disk.Label
-							"SizeIn$unit" = $total
-							"FreeIn$unit" = $free
-							PercentFree = $percentfree
-							BlockSize = $disk.BlockSize
-							IsSqlDisk = $sqldisk
-						}
-					}
-					else
-					{
-						$alldisks += [PSCustomObject]@{
-							Server = $server
-							Name = $diskname
-							Label = $disk.Label
-							"SizeIn$unit" = $total
-							"FreeIn$unit" = $free
-							PercentFree = $percentfree
-							BlockSize = $disk.BlockSize
-						}
-					}
+					$total = [math]::round($disk.Capacity / $measure, 2)
+					$free = [math]::round($disk.Freespace/$measure, 2)
+					$percentfree = [math]::round(($disk.Freespace / $disk.Capacity) * 100, 2)
+
+					$diskinfo = [PSCustomObject]@{
+						Server = $server
+						Name = $diskname
+						Label = $disk.Label
+						"SizeIn$unit" = $total
+						"FreeIn$unit" = $free
+						PercentFree = $percentfree
+						BlockSize = $disk.BlockSize
+                    }
+
+                    if ($CheckForSql -or $Detailed)
+                    {
+                        Add-Member -InputObject $diskinfo -MemberType Noteproperty IsSqlDisk -value $sqldisk
+                    }
+
+                    if ($FileSystem -or $Detailed)
+                    {
+                        Add-Member -InputObject $diskinfo -MemberType Noteproperty FileSystem -value $disk.FileSystem
+                    }
+
+                    if ($CheckFragmentation)
+                    {
+                        Add-Member -InputObject $diskinfo -MemberType Noteproperty FileSystem -value $disk.FileSystem
```

Comment:
Copy paste is not your friend today? :)

This line causes an error, as you're trying to add a member which already exists from the previous line. Should this be something like:

Add-Member -InputObject $diskinfo -MemberType Noteproperty Fragmentation -value $disk.FilePercentFragmentation